drivers/video/ov2640.c

```diff
@@ -152,17 +152,114 @@ LOG_MODULE_REGISTER(video_ov2640, CONFIG_VIDEO_LOG_LEVEL);
 #define REG32_UXGA          0x36
 
 /* Configuration arrays */
-#define SVGA_HSIZE     (800)
-#define SVGA_VSIZE     (600)
-
-#define UXGA_HSIZE     (1600)
-#define UXGA_VSIZE     (1200)
+#define SVGA_HSIZE (800UL)
+#define SVGA_VSIZE (600UL)
+
+#define UXGA_HSIZE (1600UL)
+#define UXGA_VSIZE (1200UL)
+
+#define VAL_SET(x, mask, rshift, lshift) ((((x) >> rshift) & mask) << lshift)
```

Comment:
Is this different from [`FIELD_PREP()`](https://docs.zephyrproject.org/latest/doxygen/html/group__sys-util.html#gaa03c8b31bf67a097dd9f8153a04ef86b)?

Reply:
There is no shift argument, but the macros can understand where the mask starts and ends, and shifts the value accordingly.

> representing its field **position** and width [[1]](https://docs.zephyrproject.org/latest/doxygen/html/group__sys-util.html#gaa03c8b31bf67a097dd9f8153a04ef86b)

To pre-compute the mask, you can use `GENMASK(lsb, msb)`, these two are typically used together and allow to use the value from the datasheet without risk of typo by manual tweaking the masks:

```c
#define THIS_FIELD_SET(x) FIELD_PREP(GENMASK(7, 3), x)
```

Or even better IMHO, just a plain transcription of the datasheet data without any smart:

```c
#define THIS_FIELD_MASK GENMASK(7, 3)
```

Then in the code:

```c
    {THIS_ADDR, FIELD_PREP(THIS_FIELD_MASK, 323)},
```